cfme/fixtures/vm.py

```diff
@@ -68,3 +70,33 @@ def full_template_vm_modscope(setup_provider_modscope, request, full_template_mo
 def full_template_vm(setup_provider, request, full_template, provider):
     vm_name = _get_vm_name(request)
     return _create_vm(request, full_template, provider, vm_name)
+
+
+def _create_instance(appliance, provider, template_name):
+    instance = appliance.collections.cloud_instances.instantiate(random_vm_name('pwr-c'),
+                                                                 provider,
+                                                                 template_name)
+    if not instance.exists_on_provider:
+        instance.create_on_provider(allow_skip="default", find_in_cfme=True)
+    elif instance.provider.one_of(EC2Provider) and instance.mgmt.state == VmState.DELETED:
+        instance.mgmt.rename('test_terminated_{}'.format(fauxfactory.gen_alphanumeric(8)))
+        instance.create_on_provider(allow_skip="default", find_in_cfme=True)
```

Comment:
Can you add a comment about why these checks are necessary?

Reply:
I don't really know why these checks are needed. I just moved this fixture from local to global, didn't change it and I wouldn't want to break some functionality by removing them.

Reply:
Shouldn't break functionality by removing them but since you're just moving them around I leave it as _optional_